src/components/Register/Register.js

```diff
@@ -0,0 +1,175 @@
+        passwordAgain: '',
+      },
+      check: {
+        email: '',
+        fullName: '',
+        password: '',
+      },
+      status: {
+        email: '',
+        fullName: '',
+        password: '',
+        registration: false,
+      },
+      visible: false,
+      msg: '',
+    };
+  }
+
+    handleChangeInput = ({ target: { name, value } }) => {
+      this.setState({ data: { ...this.state.data, [name]: value } });
+    };
+
+    checkEmail = async ({ target: { value: email } }) => {
+      const exp = /@/;
+      const isUnique = await isEmailUnique({ email });
+      const result = ((email.search(exp) !== -1 && isUnique));
+      this.setState({
+        status: { ...this.state.status, email: result },
+        check: { ...this.state.check, email: true },
+      });
+    };
+
+    checkName = ({ target: { value: name } }) => {
+      const exp = /[^A-Za-z\d]/;
+      const result = !(((exp.test(name)) || !(name)));
+      this.setState({
+        status: { ...this.state.status, fullName: result },
+        check: { ...this.state.check, fullName: true },
+      });
+    };
+
+    checkPass = (e) => {
+      const { password, passwordAgain } = this.state.data;
+      const result = ((password && passwordAgain && password === passwordAgain));
+      this.setState({
+        status: { ...this.state.status, password: result },
+        check: { ...this.state.check, password: true },
+      });
+    };
+
+    handleContinue = async (e) => {
+      const { data: { email, fullName, password }, check, status } = this.state;
+      if (check.email && check.fullName && check.password && status.email && status.fullName && status.password) {
+        e.preventDefault();
+        const { status, msg } = await addNewUser({ email, fullName, password });
+        this.setState({
+          visible: true,
+          status: { ...this.state.status, registration: status },
+          msg,
+        });
+      }
+    };
+
+    closeModalAdd = () => {
+      if (this.state.status.registration) {
+        this.setState({
+          data: {
+            email: '',
+            fullName: '',
+            password: '',
+            passwordAgain: '',
+          },
+          check: {
+            email: '',
+            fullName: '',
+            password: '',
+          },
+          status: {
+            email: '',
+            fullName: '',
+            password: '',
+            registration: false,
+          },
+          visible: false,
+          msg: '',
+        });
+      } else {
+        this.setState({
+          visible: false,
+          msg: '',
+          status: { registration: false },
+        });
+      }
+    };
+
+  render() {
+    const {
+      data, check, status, visible, msg,
+    } = this.state;
+    return (
+            <>
+            <form className='register-form'>
+                <p className='login-form-title'>Register</p>
+
+                <label className='login-form-label'>Email <span>(unique)</span></label>
```

Comment:
these label-input-status groups could go to a separate common component as they have common styles and functionality.